RecoEgamma/PhotonIdentification/plugins/cuts/PhoAnyPFIsoWithEAAndQuadScalingCut.cc

```diff
@@ -0,0 +1,175 @@
+#include "PhysicsTools/SelectorUtils/interface/CutApplicatorWithEventContentBase.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "RecoEgamma/EgammaTools/interface/EffectiveAreas.h"
+
+
+class PhoAnyPFIsoWithEAAndQuadScalingCut : public CutApplicatorWithEventContentBase {
+public:
+  PhoAnyPFIsoWithEAAndQuadScalingCut(const edm::ParameterSet& c);
+  
+  result_type operator()(const reco::PhotonPtr&) const override final;
+
+  void setConsumes(edm::ConsumesCollector&) override final;
+  void getEventContent(const edm::EventBase&) override final;
+
+  double value(const reco::CandidatePtr& cand) const override final;
+  
+  CandidateType candidateType() const override final { 
+    return PHOTON; 
+  }
+
+private:
+  // Cut values
+  float _C1_EB;
+  float _C2_EB;
+  float _C3_EB;
+  float _C1_EE;
+  float _C2_EE;
+  float _C3_EE;
+  // Configuration
+  float _barrelCutOff;
+  bool  _useRelativeIso;
+  // Effective area constants
+  EffectiveAreas _effectiveAreas;
+  // The isolations computed upstream
+  edm::Handle<edm::ValueMap<float> > _anyPFIsoMap;
+  // The rho
+  edm::Handle< double > _rhoHandle;
+
+  constexpr static char anyPFIsoWithEA_[] = "anyPFIsoWithEA";
+  constexpr static char rhoString_     [] = "rho";
+};
+
+constexpr char PhoAnyPFIsoWithEAAndQuadScalingCut::anyPFIsoWithEA_[];
+constexpr char PhoAnyPFIsoWithEAAndQuadScalingCut::rhoString_[];
+
+DEFINE_EDM_PLUGIN(CutApplicatorFactory,
+		  PhoAnyPFIsoWithEAAndQuadScalingCut,
+		  "PhoAnyPFIsoWithEAAndQuadScalingCut");
+
+PhoAnyPFIsoWithEAAndQuadScalingCut::PhoAnyPFIsoWithEAAndQuadScalingCut(const edm::ParameterSet& c) :
+  CutApplicatorWithEventContentBase(c),
+  _C1_EB(c.getParameter<double>("C1_EB")),
+  _C2_EB(c.getParameter<double>("C2_EB")),
+  _C3_EB(c.getParameter<double>("C3_EB")),
+  _C1_EE(c.getParameter<double>("C1_EE")),
+  _C2_EE(c.getParameter<double>("C2_EE")),
+  _C3_EE(c.getParameter<double>("C3_EE")),
+  _barrelCutOff(c.getParameter<double>("barrelCutOff")),
+  _useRelativeIso(c.getParameter<bool>("useRelativeIso")),
+  _effectiveAreas( (c.getParameter<edm::FileInPath>("effAreasConfigFile")).fullPath())
+{
+  
+  edm::InputTag maptag = c.getParameter<edm::InputTag>("anyPFIsoMap");
+  contentTags_.emplace(anyPFIsoWithEA_,maptag);
+
+  edm::InputTag rhoTag = c.getParameter<edm::InputTag>("rho");
+  contentTags_.emplace(rhoString_,rhoTag);
+
+}
+
+void PhoAnyPFIsoWithEAAndQuadScalingCut::setConsumes(edm::ConsumesCollector& cc) {
+  auto anyPFIsoWithEA = 
+    cc.consumes<edm::ValueMap<float> >(contentTags_[anyPFIsoWithEA_]);
+  contentTokens_.emplace(anyPFIsoWithEA_,anyPFIsoWithEA);
+
+  auto rho = cc.consumes<double>(contentTags_[rhoString_]);
+  contentTokens_.emplace(rhoString_, rho);
+}
+
+void PhoAnyPFIsoWithEAAndQuadScalingCut::getEventContent(const edm::EventBase& ev) {  
+  ev.getByLabel(contentTags_[anyPFIsoWithEA_],_anyPFIsoMap);
+  ev.getByLabel(contentTags_[rhoString_],_rhoHandle);
```

Comment:
You cannot call consumes in FWLite and thus getbytoken cannot work. Likewise the accessor's are not available in ```edm::EventBase``` and cannot be used in this function.